rs/p2p/state_sync_manager/src/ongoing/chunks_to_download.rs

```diff
@@ -0,0 +1,37 @@
+use ic_interfaces::p2p::state_sync::ChunkId;
+
+/// Maintains a list of chunks that still need to be downloaded
+pub(crate) struct ChunksToDownload {
+    chunks: Vec<ChunkId>,
+}
+
+impl ChunksToDownload {
+    pub(crate) fn new() -> Self {
+        Self { chunks: vec![] }
+    }
+
+    // Add chunks to the chunks to download list
+    pub(crate) fn add_chunks(&mut self, chunks: impl Iterator<Item = ChunkId>) -> usize {
+        let mut added = 0;
+        for chunk_id in chunks {
+            self.chunks.push(chunk_id);
+            added += 1;
+        }
+
+        added
```

Comment:
This could be simplified slightly (I think), but both versions work for me

```suggestion
        let initial_len = self.chunks.len();
        self.chunks.extend(chunks);
        self.chunks.len() - initial_len
```